app/src/main/java/com/epfl/drawyourpath/mainpage/fragments/DrawFragment.kt

```diff
@@ -1,34 +1,148 @@
 
+        if (savedInstanceState != null) {
+            lastKnownLocation = savedInstanceState.getParcelable(KEY_LOCATION)
+        }
+        Places.initialize(this.requireActivity().applicationContext, getString(R.string.google_api_key))
+        fusedLocationProviderClient = LocationServices.getFusedLocationProviderClient(this.requireActivity())
+
         val mapFragment = childFragmentManager
             .findFragmentById(R.id.fragment_draw_map) as SupportMapFragment
         mapFragment.getMapAsync(this)
     }
 
     override fun onMapReady(map: GoogleMap) {
-        val coordinates = LatLng(46.5185, 6.56177)
-        map.addMarker(
-            MarkerOptions()
-                .position(coordinates)
-                .title("EPFL"),
-        )
-        map.moveCamera(CameraUpdateFactory.newLatLng(coordinates))
-        map.animateCamera(CameraUpdateFactory.newLatLngZoom(coordinates, 15F))
+        this.map = map
+        map?.moveCamera(CameraUpdateFactory.newLatLngZoom(DEFAULT_LOCATION, DEFAULT_ZOOM))
+        getLocationPermission()
+        updateLocationUI()
+        getDeviceLocation()
+    }
+
+    private fun getLocationPermission() {
+        if (ContextCompat.checkSelfPermission(
+                this.requireActivity().applicationContext,
+                Manifest.permission.ACCESS_FINE_LOCATION)
+            == PackageManager.PERMISSION_GRANTED) {
+            locationPermissionGranted = true
+        } else {
+            ActivityCompat.requestPermissions(
+                this.requireActivity(),
+                arrayOf(Manifest.permission.ACCESS_FINE_LOCATION),
+                PERMISSIONS_REQUEST_ACCESS_FINE_LOCATION)
+        }
+    }
+
+    override fun onRequestPermissionsResult(requestCode: Int,
+                                            permissions: Array<String>,
+                                            grantResults: IntArray) {
+        locationPermissionGranted = false
+        when (requestCode) {
+            PERMISSIONS_REQUEST_ACCESS_FINE_LOCATION -> {
+                // If request is cancelled, the result arrays are empty.
+                if (grantResults.isNotEmpty() &&
+                    grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+                    locationPermissionGranted = true
+                }
+            }
+            else -> super.onRequestPermissionsResult(requestCode, permissions, grantResults)
+        }
+        updateLocationUI()
+    }
+
+    @SuppressLint("MissingPermission")
+    private fun updateLocationUI() {
+        if (map == null) {
+            return
+        }
+        try {
+            if (locationPermissionGranted) {
+                map?.isMyLocationEnabled = true
+                map?.uiSettings?.isMyLocationButtonEnabled = true
+            } else {
+                map?.isMyLocationEnabled = false
+                map?.uiSettings?.isMyLocationButtonEnabled = false
+                lastKnownLocation = null
+                getLocationPermission()
```

Comment:
I think we have an indirect infinite loop `getLocationPermission()` -> `onRequestPermissionsResult()` -> `updateLocationUI()` -> `getLocationPermission()` -> ...

From my understanding, permission request is asked directly on creation so we don't need to ask it again here.

Reply:
Yes we do have an infinite loop, at first I thought it was intended but after modifying the code, it is clearly not.
Simple fix, remove the getLocationPermission() call in updateLocationUI()